### PR TITLE
Tag Lora v0.5.1

### PR DESCRIPTION
**Pull request by scidom**
Ported accept-reject sampler (ARS), random adaptive Metropolis (RAM), HMC and SMMALA from older Lora and added support for forward and reverse mode automatic differentiation of targets.

### CODE REVIEW

File: Lora/versions/0.5.1/requires [hunk 1]
```diff
@@ -0,0 +1,7 @@
+julia 0.4-
```
Comment (tkelman) [inline Lora/versions/0.5.1/requires]: drop the trailing `-`

Comment (scidom) [inline Lora/versions/0.5.1/requires]: Thanks @tkelman, I dropped it.

Comment (tkelman) [inline Lora/versions/0.5.1/requires]: here too https://github.com/scidom/METADATA.jl/blob/pull-request/f7d14f72/Lora/versions/0.5.1/requires

**Comment by scidom** [comment]
I think I changed it in metadata too now @tkelman.

**Pull request merged**